### DISCUSSION TITLE
Automated Email Bug-Report

### DISCUSSION THREAD

**Discussion by Cryptec** [firefly-iii/firefly-iii#4367] [Firefly III]
**Bug description**
I am running Firefly III version v5.4.6 and my problem is:

 file_put_contents(/var/www/firefly-iii/storage/framework/cache/data/22/0a/220a6b7246108a400ddb0ca65d3b482393dc3d43): failed to open stream: No such file or directory.

**Extra info**

i receive this via an automatied Firefly 3 email several times when i work with Transactions, but everything seems to work.

**Stacktrace**

#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError()
#1 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php(185): file_put_contents()
#2 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php(74): Illuminate\Filesystem\Filesystem->put()
#3 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Cache/FileStore.php(152): Illuminate\Cache\FileStore->put()
#4 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Cache/Repository.php(355): Illuminate\Cache\FileStore->forever()
#5 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Cache/CacheManager.php(396): Illuminate\Cache\Repository->forever()
#6 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Illuminate\Cache\CacheManager->__call()
#7 /var/www/firefly-iii/app/Support/CacheProperties.php(95): Illuminate\Support\Facades\Facade::__callStatic()
#8 /var/www/firefly-iii/app/Helpers/Report/NetWorth.php(136): FireflyIII\Support\CacheProperties->store()
#9 /var/www/firefly-iii/app/Api/V1/Controllers/SummaryController.php(382): FireflyIII\Helpers\Report\NetWorth->getNetWorthByCurrency()
#10 /var/www/firefly-iii/app/Api/V1/Controllers/SummaryController.php(119): FireflyIII\Api\V1\Controllers\SummaryController->getNetWorthInfo()
#11 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): FireflyIII\Api\V1\Controllers\SummaryController->basic()
#12 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\Routing\Controller->callAction()
#13 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(255): Illuminate\Routing\ControllerDispatcher->dispatch()
#14 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(197): Illuminate\Routing\Route->runController()
#15 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(691): Illuminate\Routing\Route->run()
#16 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\Routing\Router->Illuminate\Routing\{closure}()
#17 /var/www/firefly-iii/app/Api/V1/Controllers/SummaryController.php(96): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#18 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): FireflyIII\Api\V1\Controllers\SummaryController->FireflyIII\Api\V1\Controllers\{closure}()
#19 /var/www/firefly-iii/app/Api/V1/Controllers/Controller.php(64): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#20 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(149): FireflyIII\Api\V1\Controllers\Controller->FireflyIII\Api\V1\Controllers\{closure}()
#21 /var/www/firefly-iii/app/Http/Middleware/Binder.php(79): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#22 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\Http\Middleware\Binder->handle()
#23 /var/www/firefly-iii/app/Http/Middleware/Authenticate.php(74): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#24 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\Http\Middleware\Authenticate->handle()
#25 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#26 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(693): Illuminate\Pipeline\Pipeline->then()
#27 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(668): Illuminate\Routing\Router->runRouteWithinStack()
#28 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(634): Illuminate\Routing\Router->runRoute()
#29 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(623): Illuminate\Routing\Router->dispatchToRoute()
#30 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(166): Illuminate\Routing\Router->dispatch()
#31 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}()
#32 /var/www/firefly-iii/app/Http/Middleware/InstallationId.php(53): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#33 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\Http\Middleware\InstallationId->handle()
#34 /var/www/firefly-iii/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#35 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fideloper\Proxy\TrustProxies->handle()
#36 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#37 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
#38 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#39 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\TransformsRequest->handle()
#40 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#41 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\ValidatePostSize->handle()
#42 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(87): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#43 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\Foundation\Http\Middleware\PreventRequestsDuringMaintenance->handle()
#44 /var/www/firefly-iii/app/Http/Middleware/SecureHeaders.php(51): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#45 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): FireflyIII\Http\Middleware\SecureHeaders->handle()
#46 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}()
#47 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(141): Illuminate\Pipeline\Pipeline->then()
#48 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(110): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter()
#49 /var/www/firefly-iii/public/index.php(76): Illuminate\Foundation\Http\Kernel->handle()
#50 {main}

**Comment by JC5** [comment]
- https://docs.firefly-iii.org/firefly-iii/faq/self_hosted/#unable-to-write-to-cache-directory
- https://docs.firefly-iii.org/firefly-iii/faq/docker/#i-get-permission-denied-errors-on-the-cache-folder

**Reply by Cryptec** [reply]
looks like this did the tick, thanks!

**Reply by JC5** [reply]
Glad to hear it! 😅
